### ISSUE TITLE
Separate "modulo" and "remainder" nodes

### ISSUE THREAD

**Issue by iamgreaser** [Neos-Metaverse/NeosPublic#3169]
### Is your feature request related to a problem? Please describe.

Please remind me if there is a proper modulo node and not just a truncation remainder node, I don't have Neos open right now but if I recall correctly it only provides a truncation remainder. Anyway, here comes the issue.

There is a common set of workarounds needed whenever a programming language makes the mortal sin of providing a `%` operator which does truncation remainder instead of modulo. It is likely that LogiX did whatever C# did, which may be based on whatever C did and/or whatever the relevant CPUs ended up doing because reasons. Either way, LogiX commits said mortal sin.

In general, the only way to sanely use the `%` node at the moment is in the form of `(((x % D) + D) % D)` unless you want things to break whenever the `x` input goes negative.

Remainder and modulo do the same thing when both inputs are positive, but as soon as a negative numerator input makes its way in, they differ - modulo continues the pattern intuitively, while truncating remainder starts moving into negatives and ultimately creates a discontinuity in terms of the period.

Because the `%` node already exists, we cannot simply change it to use modulo, as it might actually break something, despite probably fixing a lot of other things.

### Relevant issues

Nothing seems to directly address this, but implementing this as-is will make some progress towards #795 which pertains to giving intuitive names for various nodes.

### Describe the solution you'd like

Add a "Mod" node, short for "Modulus", and place it under Operators.

Rename the `%` node to "Rem", short for "Remainder". If someone actually manages to successfully find a valid reason for using this then they can choose it explicitly.

Or maybe rename it to "Truncating Remainder" and bury it somewhere under Math, because I can guarantee it's not what you want.

### Describe alternatives you've considered

The usual `(((x % D) + D) % D)` workaround as mentioned above.

Alternatively, based on the definition of a remainder, one can do `x - floor(x / D)`, but I suspect the numerical accuracy is worse.

### Additional context

Languages which have both truncation remainder and modulus under explicit names include Cobol, Common Lisp, Scheme. Cobol and Common Lisp both call these `rem` and `mod`, while Scheme uses the longer names of `remainder` and `modulo`.

Languages which present `%` as modulus include Lua, Perl 5, Pure Data, Python, Raku, Ruby, Tcl 8.6. Emacs Lisp presents it as `mod`, and appears to not have truncation remainder.

Truncation *division* does have valid applications for when one is multiplying against a vector magnitude or a speed or a velocity, but I have never seen a valid application for truncation *remainder*.

Floor division and modulo, on the other hand, are often used with grids to compute cell position and offset within a cell, and modulo is often used for periodic stuff.

**Comment by iamgreaser** [comment]
...what? That's actually the modulo node?

OK, I think this issue's now fundamentally about renaming and rearranging two nodes which exist.

**Comment by mralext20** [comment]
re: renaming etc, duplicate of #795 ?

**Comment by iamgreaser** [comment]
Possibly... I'll weigh in on the "Repeat" node, but the remaining aspect which I'm still not entirely sure about the "duplicateness" of is that I feel that Neos ends up encouraging people to use the wrong remainder node.

**Comment by iamgreaser** [comment]
> Similarly % is the same modulo operation that you find in a number of programming languages as well - it's how it's commonly defined and we do not want to go against standard there.

No, I have to correct this. `%` is one of two closely-related operations. It depends entirely on the language as to which one it is. Unless adequately defined in the documentation, there is no way to know which one of these without either trying to use it, or trying to do a division which produces an integer. Hence why I am stressing that this should be renamed. Typically it'd be "Remainder" or "Rem".

* Languages where `%` is modulus: Lua, Perl 5, Python, Raku, Ruby, Tcl 8.6
* Languages where `%` is truncating remainder: Java, LogiX, PHP 7, probably C#
* Languages where which one it is may be undefined behaviour, I'm actually unsure but it tends to be truncating remainder: C, C++, Pure Data (deprecated in favour of `mod`)

I messed up when I said Pure Data uses `%` to denote modulo, it actually uses `mod`, and says "In addition the `%` operator (provided for back compatibility) is like `mod` but acts differently for negative inputs (and might act variously depending on CPU design).".
